File: solucao-csharp/Exercises/Ex003.cs
namespace desafio_target.Exercises
{
    internal class ExerciseThree
    {
        public static void ShowAnswer()
        {
            Console.Write("A) ");
            for (int i = 1; i <= 9; i += 2)
            {
                Console.Write(i);

                if (i < 9)
                {
                    Console.Write(", ");
                }
                else
                {
                    Console.Write("\n");
                }
            }

            Console.Write("B) ");
            for (int i = 1; i <= 7; i++)
            {
                double answer = Math.Pow(2, i);

                Console.Write(answer);

                if (i < 7)
                {
                    Console.Write(", ");
                }
                else
                {
                    Console.Write("\n");
                }
            }

            Console.Write("C) ");
            for (int i = 0; i <= 7; i++)
            {
                double answer = Math.Pow(i, 2);

                Console.Write(answer);

                if (i < 7)
                {
                    Console.Write(", ");
                }
                else
                {
                    Console.Write("\n");
                }
            }

            Console.Write("D) ");
            for (int i = 2; i <= 10; i += 2)
            {
                double answer = Math.Pow(i, 2);

                Console.Write(answer);

                if (i < 10)
                {
                    Console.Write(", ");
                }
                else
                {
                    Console.Write("\n");
                }
            }

            Console.Write("E) ");
            int[] fibonacci = new int[8];
            int start = 0;
            int next = 1;

            for (int i = 0; i < 8; i++)
            {
                fibonacci[i] = start;
                int temp = next;
                next += start;
                start = temp;
            }

            Console.WriteLine(string.Join(", ", fibonacci));

            Console.Write("F) ");
            Dictionary<int, string> numerosPorExtenso = new()
            {
                {2, "Dois"},
                {10, "Dez"},
                {12, "Doze"},
                {16, "Dezesseis"},
                {17, "Dezessete"},
                {18, "Dezoito"},
                {19, "Dezenove"},
                {200, "Duzentos"}
            };

            Console.Write(string.Join(", ", numerosPorExtenso.Keys));
            Console.Write("\n");
            Console.Write(string.Join(", ", numerosPorExtenso.Values));
            Console.WriteLine();
        }
    }
}

File: solucao-node-ts/src/ex003.ts
class ExerciseThree {
    static showAnswer() {
      // A
      console.log("A) " + Array.from({ length: 5 }, (_, i) => 2 * i + 1).join(", "));
  
      // B
      console.log("B) " + Array.from({ length: 7 }, (_, i) => Math.pow(2, i + 1)).join(", "));
  
      // C
      console.log("C) " + Array.from({ length: 8 }, (_, i) => Math.pow(i, 2)).join(", "));
  
      // D
      console.log("D) " + Array.from({ length: 5 }, (_, i) => Math.pow(2 * (i + 1), 2)).join(", "));
  
      // E
      const fibonacci = this.generateFibonacci(8);
      console.log("E) " + fibonacci.join(", "));
  
      // F
      const numerosPorExtenso = {
        2: "Dois",
        10: "Dez",
        12: "Doze",
        16: "Dezesseis",
        17: "Dezessete",
        18: "Dezoito",
        19: "Dezenove",
        200: "Duzentos",
      };
  
      console.log("F) " + Object.keys(numerosPorExtenso).join(", "));
      console.log(Object.values(numerosPorExtenso).join(", "));
    }
  
    static generateFibonacci(length) {
      let fibonacci = [0, 1];
  
      while (fibonacci.length < length) {
        const nextTerm = fibonacci.slice(-1)[0] + fibonacci.slice(-2, -1)[0];
        fibonacci.push(nextTerm);
      }
  
      return fibonacci;
    }
  }
  
  export default ExerciseThree;